### PR TITLE
feat: implement submit application endpoint

### PR DESCRIPTION
**Pull request by inderps**
## 🚀 Pull Request Checklist

- **Title:** 
 - implement submit application endpoint

- **Description:**
  - Implement new endpoint, '/api/v1/applications/{id}/submit'.
  - Integrate with REMS.

### CODE REVIEW

File: src/main/java/io/github/genomicdatainfrastructure/daam/services/SubmitApplicationService.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    public void submitApplication(String id) {
+        var principal = (OidcJwtCallerPrincipal) identity.getPrincipal();
+        String userId = principal.getClaim(USER_ID_CLAIM);
+
+        SubmitApplicationCommand command = SubmitApplicationCommand.builder()
+                .applicationId(Long.valueOf(id))
+                .build();
+
+        remsApplicationCommandApi.apiApplicationsSubmitPost(command, remsApiKey, userId);
```
Comment (inderps) [inline src/main/java/io/github/genomicdatainfrastructure/daam/services/SubmitApplicationService.java:45]: Only submitting applications that are in DRAFT or RETURNED status will be implemented in next PR

**Review by EmiPali — APPROVED** [review]

File: src/main/java/io/github/genomicdatainfrastructure/daam/services/SubmitApplicationService.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import io.github.genomicdatainfrastructure.daam.remote.rems.api.RemsApplicationCommandApi;
+import io.github.genomicdatainfrastructure.daam.remote.rems.model.SubmitApplicationCommand;
+import io.quarkus.oidc.runtime.OidcJwtCallerPrincipal;
+import io.quarkus.security.identity.SecurityIdentity;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import java.util.List;
```
Comment (brunopacheco1) [inline src/main/java/io/github/genomicdatainfrastructure/daam/services/SubmitApplicationService.java:17]: hint: `List` is not used anywhere.

**Review by brunopacheco1 — DISMISSED** [review]

File: src/main/java/io/github/genomicdatainfrastructure/daam/services/SubmitApplicationService.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        SecurityIdentity identity,
+        @RestClient RemsApplicationCommandApi applicationsApi
+    ) {
+        this.remsApiKey = remsApiKey;
+        this.identity = identity;
+        this.remsApplicationCommandApi = applicationsApi;
+    }
+
+    public void submitApplication(String id) {
+        var principal = (OidcJwtCallerPrincipal) identity.getPrincipal();
```
Comment (brunopacheco1) [inline src/main/java/io/github/genomicdatainfrastructure/daam/services/SubmitApplicationService.java:38]: suggestion: Can you please move the resolution of `userId` to the controller/api class? 

`OidcJwtCallerPrincipal` is part of authentication/authorization layer, whereas use case services must be agnostic of this kind of implementation details - imagine we want to support multiple authentication mechanisms.

Can you also please fix `CreateApplicationService`? I just missed this problem there as well.

Comment (inderps) [reply src/main/java/io/github/genomicdatainfrastructure/daam/services/SubmitApplicationService.java:38]: I will do that in separate PR if thats okay with you

Comment (brunopacheco1) [reply src/main/java/io/github/genomicdatainfrastructure/daam/services/SubmitApplicationService.java:37]: Sure, that's why I approved. I did not close the conversation just to ensure you get to read it.

**Review by brunopacheco1 — APPROVED** [review]

**Pull request merged**